README.md

```diff
@@ -1,70 +1,118 @@
-MuSig2.
+## Schnorr Signature of libsecp256k1
 
-The aggregated public key is `X = X_1 * a_1 + ... + X_n * a_n` and _R_ is `R = R_1 * b^(j-1) + ... + R_V * b^(V-1)`.
+The library offers an optional module implementing Schnorr signatures over the curve `secp256k1`. 
+The scheme is compatible with [BIP-340](https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki) that produces 64-byte Schnorr signatures. 
+BIP-340 standardized the scheme with some modifications for practical purposes.
 
-The verification of MuSig2 checks whether:
 
-```
-    G * sig = R + X * c.
-```
+### Modifications
 
-Therefore, we made a little tweak to make sure MuSig2 can be verified by `secp256k1_schnorrsig_verify`.
+- **Encoding nonce and public key:**
+Instead of encoding full `X`, `Y` coordinates of $R$ and $X$ (64-byte public key, 96-byte signature), or compressed encoding (33-byte public key, 65-byte signature), BIP-430 preferred to use xonly encoding (32-byte public key, 64-byte signature).
 
-1. If _X_ has an odd y coordinate and _R_ has even: **Negate _c_**
+- **Implicit `Y` coordinate:**
+Schnorr signatures of BIP-430 implicitly choose the `Y` coordinate that is even.
 
-2. If _R_ has an odd y coordinate and _X_ has even: **Negate every element in _b_LIST_ and recompute _R_.**
+- **Tagged hashes:**
+In order to prevent the collisions and reuse of nonce, it is preferred to include the tag by prefixing the hash data with `SHA256(tag) || SHA256(tag)`.
 
-3. If both _X_ and _R_ have odd y coordinates: **Negate aggregated signature _agg_sig_.** 
+### Security considerations
 
-___
+The xonly encoding of elliptic curve points causes ambiguity since every `X` coordinate has two possible `Y` coordinates. 
+To avoid this ambiguity they chose to select elliptic curve points with even `Y` coordinates.
+The xonly encoding of a point prefixed by the byte `0x02` is equivalent to representing it with a 33-byte but more compact. 
+Moreover, this encoding does not reduce the security of the system. 
+Even if there is an algorithm to solve ECDLP for the xonly encoding, then the full encoding will also be automatically broken since the `X` coordinate has two possible `Y` coordinates as negative and positive.
 
-### Progress
+## Adapting MuSig2 to PIB-430
 
-- [x] Update the function and parameter namings.
-- [x] Prevent reuse of r values.
-- [x] Create musig2 context to simplify the API.
-- [ ] Tests (Valgrind).
+[MuSig2](https://eprint.iacr.org/2020/1261.pdf) 
+is a two-round multi-signature scheme that outputs an ordinary Schnorr signature.
+
+$$ G \cdot s = R + X \cdot c. $$
 
+The aggregated public key $X$ is generated with the public keys of all signers:
 
+$$ X = \Sigma_{i = 1}^{N} (X_i \cdot a_i) $$
 
+where $a_i$ is the key aggregate coefficient.
+The commitment $R$ is also generated with the commitments of signers and the nonce $b$.
 
+$$ R = \Sigma_{i = 1}^{V}(R_j^{b^{j-1}}). $$
```

Comment:
You are using non-introduced notation. Everything should be introduced, so maybe it makes sense to include all the steps of MuSig2 as introduced in the paper.